### ISSUE TITLE
NestingDepth takes into consideration BEM selectors

### ISSUE THREAD

**Issue by cdreve** [brigade/scss-lint#810]
Hi all,

With the current version, is it possible for the `NestingDepth` to consider BEM selectors as nesting depth 1? If not, could this be added?

**Current config:**
```bash
 ~ scss-lint -v
scss-lint 0.49.0
```
```yml
NestingDepth:
  enabled: true
  max_depth: 1
```

**SCSS:**
```scss
.block {
  color: red;
  
    &__element {
      color: blue;
    }
}
```

**Expected result:**
In this case I would expect the NestingDepth to be 1, since this generates 2 CSS selectors that don't have any selectors within.
```scss
.block {color: red;}
.block__element {color: blue;}
```

**Actual result:** 
`NestingDepth: Nesting should be no greater than 1, but was 2`

**Note:** For linting BEM selectors we already have `BemDepth`.

Thanks

**Comment by calazans10** [comment]
I have the same issue, @cdreve .

**SCSS:**
```sass
.block {
   ...
  &__element {
    ...
    &--modifier {
      ...
      &::before {
        ...
      }
    }
}
```

**Result**

```css
  .block {}
  .block__element {}
  .block__element-modifier {}
  .block__element-modifier::before {}
```

**Warning**

```NestingDepth: Nesting should be no greater than 3, but was 4```

**Comment by sds** [comment]
Hey all, 

`NestingDepth` is working as designed here. Its intention is to notify you when you are nesting (i.e. indenting) more than a given maximum.

Having deeply-nested/indented code is a smell once you start working with larger amounts of code within the nested blocks. You won't be able to easily tell what rule you are nested within without scrolling back up:

```scss
.block {
  &__element {
    ... 200 lines ...
    &--modifier {
      &::before {
        ...what element is this modifier applying to again?
      }
    }
  }
}
```

My recommendation if you don't find this linter useful is to disable it in this case. I'm of course open to dialog on this, but even then a pull request would speak louder than words. Thanks!
